tests/unit/multi-currency/test-class-compatibility.php

```diff
@@ -40,23 +50,220 @@ public function setUp() {
 		$this->mock_coupon = $this->createMock( \WC_Coupon::class );
 	}
 
+	/**
+	 * @dataProvider woocommerce_filter_provider
+	 */
+	public function test_registers_woocommerce_filters_properly( $filter, $function_name ) {
+		$priority = has_filter( $filter, [ $this->compatibility, $function_name ] );
+		$this->assertGreaterThan(
+			10,
+			$priority,
+			"Filter '$filter' was not registered with '$function_name' with a priority higher than the default."
+		);
+		$this->assertLessThan(
+			100,
+			$priority,
+			"Filter '$filter' was registered with '$function_name' with a priority higher than than 100, which can cause double conversions."
+		);
+	}
+
+	public function woocommerce_filter_provider() {
+		return [
+			[ 'option_woocommerce_subscriptions_multiple_purchase', 'maybe_disable_mixed_cart' ],
+			[ 'woocommerce_subscriptions_product_price', 'get_subscription_product_price' ],
+			[ 'woocommerce_product_get__subscription_sign_up_fee', 'get_subscription_product_signup_fee' ],
+			[ 'woocommerce_product_variation_get__subscription_sign_up_fee', 'get_subscription_product_signup_fee' ],
+		];
+	}
+
+	public function test_get_subscription_product_price_does_not_convert_price() {
+		$this->mock_utils->method( 'is_call_in_backtrace' )->willReturn( true );
+		$this->mock_wcs_cart_contains_renewal( true );
+		$this->assertSame( 10.0, $this->compatibility->get_subscription_product_price( 10.0, $this->mock_product ) );
+	}
+
+	public function test_get_subscription_product_price_converts_price_test_1() {
```

Comment:
Would it be possible add a docblock comment for these tests just saying what each individual scenario is checking for, like you've done with the `test_get_subscription_product_signup_fee_does_not_convert_price_*` tests?

Reply:
Totally possible, and now done. I initially did that due to I was thinking too hard about it and wanted to describe all of what was happening, looking at it today I was able to conjure up better names.